apps/src/applab/ProtectedDesignWorkspace.jsx

```diff
@@ -0,0 +1,44 @@
+import { ApplabInterfaceMode } from './constants';
+import React from 'react';
+import { connect } from 'react-redux';
+
+const ProtectedDesignWorkspace = React.createClass({
+  propTypes: {
+    interfaceMode: React.PropTypes.oneOf([
+      ApplabInterfaceMode.CODE,
+      ApplabInterfaceMode.DESIGN,
+      ApplabInterfaceMode.DATA
+    ]).isRequired
+  },
+
+  shouldComponentUpdate: function (nextProps) {
+    // This component is currently semi-protected. We don't want to completely
```

Comment:
I wonder if we're setting ourselves up for confusion by calling this `ProtectedDesignWorkspace` when protected already has a somewhat specific stricter meaning. Not sure what this concept would be. The best I can come up with is `SemiProtectedDesignWorkspace` which is quite a mouthful.